sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -83,10 +84,12 @@ trait CreateHiveTableAsSelectBase extends DataWritingCommand {
     tableDesc: CatalogTable,
     tableExists: Boolean): DataWritingCommand
 
+  def writingCommandClass: Class[_]
+
   override def argString(maxFields: Int): String = {
     s"[Database:${tableDesc.database}, " +
     s"TableName: ${tableDesc.identifier.table}, " +
-    s"InsertIntoHiveTable]"
+    s"${Utils.getSimpleName(writingCommandClass)}]"
```

Comment:
I think we could simply `Utils.getSimpleName(this.getClass)}`.

Reply:
Oh, I am sorry, it's not checking self.

Reply:
Yes. There's probably nicer ways to do this, for example make the base class type parameterized on the class of the writing command, and then here we'd just need to do `Utils.getSimpleName(classOf[T])`. But it's somewhat of an overkill to use generics here just for the explain output, so I'm opting for a simpler solution.